firedrake/projection.py

```diff
@@ -163,7 +177,21 @@ class BasicProjector(ProjectorBase):
     @cached_property
     def rhs_form(self):
         v = firedrake.TestFunction(self.target.function_space())
-        form = firedrake.inner(self.source, v)*firedrake.dx
+        F = self.target.function_space()
+        mixed = isinstance(F.ufl_element(), ufl.MixedElement)
+        if not mixed and isinstance(F.finat_element, HDivTrace):
+            if F.extruded:
+                form = (firedrake.inner(self.source, v)*firedrake.ds_t
+                        + firedrake.inner(self.source, v)*firedrake.ds_v
+                        + firedrake.inner(self.source, v)*firedrake.ds_b
+                        + firedrake.inner(self.source('+'), v('+'))*firedrake.dS_h
+                        + firedrake.inner(self.source('+'), v('+'))*firedrake.dS_v)
+            else:
+                form = (firedrake.inner(self.source, v)*firedrake.ds
+                        + firedrake.inner(self.source('+'), v('+'))*firedrake.dS)
```

Comment:
also average source here.

Reply:
We probably want to change this here 
https://github.com/firedrakeproject/firedrake/blob/1fafb01f15f817bece2cbabddd1367b2db7bc151/tests/extrusion/test_trace_extr.py#L24 and here https://github.com/firedrakeproject/firedrake/blob/1fafb01f15f817bece2cbabddd1367b2db7bc151/tests/regression/test_trace_galerkin_projection.py#L59 too?

Reply:
Ah, actually I can dismiss this, both tests define the source in a continuous space.